File: Applied Course/4.Problem Solving/1.Problems on Array/23.Count Negative Numbers in a Sorted Matrix.cpp
#include<bits/stdc++.h>
using namespace std;
// Logic-1
class Solution {
public:
    int countNegatives(vector<vector<int>>& grid) {
        int r=grid.size(),c=grid[0].size(),count=0;
        for(int i=0;i<r;i++)
            for(int j=0;j<c;j++)
                if(grid[i][j]<0)
                    count++;
        return count;
    }
};
// Logic-2
class Solution {
public:
    int countNegatives(vector<vector<int>>& grid) {
        int r=grid.size(),c=grid[0].size(),count=0;
        for(int i=r-1;i>=0;i--)
            for(int j=0;j<c;j++)
                if(grid[i][j]<0)
                {
                    count+=(c-j);
                    break;
                }
        return count;
    }
};
int main()
{
    // https://leetcode.com/problems/count-negative-numbers-in-a-sorted-matrix/
    return 0;
}

File: Applied Course/4.Problem Solving/1.Problems on Array/23.Count Negative Numbers in a Sorted Matrix.py
# https://leetcode.com/problems/count-negative-numbers-in-a-sorted-matrix/
class Solution:
    def countNegatives(self, grid: List[List[int]]) -> int:
        rows=len(grid)
        cols=len(grid[0])
        i,j=rows-1,0
        count=0
        while i>=0 and j<cols:
            if grid[i][j]<0:
                count+=(cols-j)
                i-=1
            else:
                j+=1
        return count